google/resource_google_project.go

```diff
@@ -173,6 +177,40 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
+func resourceGoogleProjectCheckPreRequisites(config *Config, d *schema.ResourceData) error {
+	ib, ok := d.GetOk("billing_account")
+	if !ok {
+		return nil
+	}
+	ba := "billingAccounts/" + ib.(string)
+	req := &cloudbilling.TestIamPermissionsRequest{
+		Permissions: []string{"billing.resourceAssociations.create"},
+	}
+	resp, err := config.clientBilling.BillingAccounts.TestIamPermissions(ba, req).Do()
+	if err != nil {
+		return fmt.Errorf("failed to check permissions on billing account %q: %v", ba, err)
+	}
+	if diff := diffStringSlices(resp.Permissions, req.Permissions); len(diff) > 0 {
```

Comment:
This is a bit overly complicated. Can you just loop through `resp.Permissions` and make sure that `billing.resourceAssociations.create` is present instead of doing this slice comparison?

Reply:
Done.